### ISSUE TITLE
Road to v2.0.0

### ISSUE THREAD

**Issue by hyrodium** [JuliaApproximation/FastGaussQuadrature.jl#129]
The following is a list of changes I intended for v1.0.0. However, since that version has already been released, these changes will be targeted for v2.0.0:

* [ ] Remove `gausschebyshev`
* [ ] Remove `besselroots`
* [ ] Organize function names etc.
  * For example, `FastGaussQuadrature.asy` should be renamed to `FastGaussQuadrature.gausslegenedre_asy`.
  * They (`gausslegendre_asy`, `gausslaguerre_asy`, etc.) will be treated as 'non-exported stable API', meaning they won't be exported but are stable for use.
  * Consider adding an `_` prefix for purely internal functions, such as `underflow_threshold`.

**Comment by dlfivefifty** [comment]
Removing deprecated functions and changing internal function names can all happen on 1.1

**Comment by jishnub** [comment]
Please don't remove deprecated functions such as `gausschebyshev` without a breaking change, as these used to be a part of the official API. By default, Julia doesn't warn about the deprecation unless one starts it with `--depwarn=yes`, so this may confuse users. Is there a pressing need to remove these functions right away anyway?

**Comment by hyrodium** [comment]
>changing internal function names can all happen on 1.1

That's right. But I was just planning to make this package more stable before releasing v1.0.0.

>Is there a pressing need to remove these functions right away anyway?

No. I believe we should not remove the exported functions before we get an agreement.
